lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -185,6 +202,36 @@ describe('NxDrawer', function() {
 
       expect(mockOnClose).toHaveBeenCalled();
     });
+
+    it('does not execute onClose when clicked outside of the drawer when closeBtnDisabled is true', async function() {
+      const user = userEvent.setup();
+      const mockOnClose = jest.fn();
+
+      render(
+        <div>
+          <div>Outside</div>
+          <NxDrawer open={true} onClose={mockOnClose} closeBtnDisabled={true}>
+            <NxButton>Inside</NxButton>
+          </NxDrawer>
+        </div>
+      );
+
+      const dialog = screen.getByRole('dialog', { hidden: true });
+      await fireEvent.animationEnd(dialog);
+
+      const outsideDiv = screen.getByText('Outside');
+      const insideButton = screen.getByRole('button', { name: 'Inside' });
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      await user.click(insideButton);
+
+      expect(mockOnClose).not.toHaveBeenCalled();
```

Comment:
Redundant with the final check
```suggestion
      await user.click(insideButton);
```